### DISCUSSION TITLE
Long issue bodies in shell script

### DISCUSSION THREAD

**Discussion by postlund** [community/community#24967] [Actions]
<p>I’m trying to create a workflow that parses raw logs (from a particular project) posted in issues, convert them to a HTML representation with a python script I’ve written and post a link back to the issue to that file. At this point, I have something that works but with a few limitations.</p>
<p>If a try to store the content body as an environment variable like this:</p>
<pre><code class="lang-yaml">...
    steps:
    - name: Generate HTML output
      id: genout
      env:
        COMMENT_BODY: ${{ toJson(github.event.comment.body) }}
      run: 
        ...
</code></pre>
<p>I will get the infamous <code>Argument list too long</code> if the body is too large as it exceeds the environment variable maximum size. Unfortunately that is common as logs can be rather large. It works as expected for small logs however.</p>
<p>Trying to stuff <code>${{ toJson(github.event.comment.body) }}</code> into <code>run</code> to circumvent that problem creates other problems as characters are interpreted by the shell, so backquotes and things like that can throw errors if they mismatch. Not good.</p>
<p>What I would prefer is a way to store the issue body in a file directly, so I can read the content from there. Is that possible somehow (without going via shell)?</p>

**Comment by chrispat** [comment, marked as answer]
<p>The full contents are already on disk.  If you look at <a href="https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context" rel="noopener nofollow ugc">Context and expression syntax for GitHub Actions - GitHub Docs</a> <code>github.event_path</code> is the location of a .json file that contains the full webhook payload. You should be able to read that file in and parse from there.</p>

**Comment by postlund** [comment]
<p>Ah, that is excellent! That will definitely work for me, thanks for point that out!</p>
